### DISCUSSION TITLE
Await google maps idle state before render

### DISCUSSION THREAD

**Discussion by ImDino** [visgl/deck.gl#6791] [Q&A] [answered]
I want my render method in my composite layer to await google maps to be idle. I tried a bunch of stuff.

I want it to work as following just to give an idea, unfortunately I can't find any google maps deckoverlay property to await the map.

If I call the renderLayers method manually I just get the sublayers, but I need them to be sent from where the renderLayers is called automatically otherwise for it to work that way I guess.

map.addListener("zoom_changed", () => {
  mapReady = false;
  console.log('zoom_changed');
});

map.addListener("idle", () => {
  mapReady = true;
  console.log('idle');
});

...

class IconClusterLayer extends CompositeLayer  {
  ...
  renderLayers() {
    await mapReady;
    return new Icon layer({...})
  }
}

**Comment by Pessimistress** [comment]
`renderLayers` is synchronous. If you wish to only draw a layer after the map is ready, construct your overlay instance with `layers: []`, and call `overlay.setProps({layers: [myClusterLayer]})` when the map is loaded.

**Reply by ImDino** [reply]
I didn't mean in the initiation stage, I meant when zooming and waiting for the map to load the tiles before my composite layer updates the index and re-renders. I managed to fix it after reading the source code though, I call layer.setNeedsUpdate() manually instead. Too bad it's not in the documentation.

**Comment by ImDino** [comment, marked as answer]
I managed to fix it after reading the source code though, I call layer.setNeedsUpdate() manually instead. Too bad it's not in the documentation.
